gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -427,6 +450,13 @@ public void apply(JobListener jobListener, JobContext jobContext)
         TimingEvent workUnitsCreationTimer =
             this.eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.WORK_UNITS_CREATION);
         Source<?, ?> source = this.jobContext.getSource();
+        if (source instanceof KafkaSource) {
```

Comment:
I think this is leaking the abstraction if we just use `instanceof` here ... AbstractJobLauncher is supposed to contain no implementation-specific constructs and I would argue even importing stuff from `import org.apache.gobblin.source.extractor.extract.kafka` package should not happen here. 

Would you reconsider this design, do you really need the abstractJobLauncher to be registered?